modules/tinymce/src/core/test/ts/browser/dom/ControlSelectionTest.ts

```diff
@@ -187,4 +188,14 @@ describe('browser.tinymce.core.dom.ControlSelectionTest', () => {
     const wrapper = UiFinder.findIn(TinyDom.body(editor), 'span.mce-preview-object').getOrDie();
     getAndAssertDimensions(wrapper, 402 + 100, 202 + 50);
   });
+
+  it('TINY-5947: data-mce-selected should be set synchronously when selecting control elements', async () => {
```

Comment:
This also checks the async nature of the resize handles so maybe either spit this up to two tests or change the description. Should it maybe also check that they are not present first sync then check if they are present after a while with the waiter.

Reply:
I probably should just have left the last line out, but I threw it in just as a double check they still do render at some stage. The main thing this is testing is the sync part and we don't really care if the resize handles are rendered immediately or in say the following event loop. So I don't really think anything should change here, but if you really want something done I'll just remove the wait for the handles at the end.